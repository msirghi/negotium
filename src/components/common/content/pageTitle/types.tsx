import { ProjectSettingsOption } from '../../../../common/constants/enums';

export type SettingsOptions = {
  settingsOptions?: {
    onClick: () => void;
  };
  projectOptions?: {
    show: boolean;
    onClick?: (option: ProjectSettingsOption) => void;
  };
};
