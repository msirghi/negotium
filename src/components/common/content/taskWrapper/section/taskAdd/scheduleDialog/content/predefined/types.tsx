import { OverridableComponent } from '@mui/types';
import { SvgIconTypeMap } from '@mui/material';

export type IPredefinedOption = {
  title: string;
  Icon: OverridableComponent<SvgIconTypeMap<{}, 'svg'>> & { muiName: string };
  code: PREDEFINED_CODES;
};

export enum PREDEFINED_CODES {
  TODAY = 'TODAY',
  TOMORROW = 'TOMORROW'
}
