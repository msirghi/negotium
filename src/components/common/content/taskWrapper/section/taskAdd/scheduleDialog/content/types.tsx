import { Nullable } from '../../../../../../../../common/types/common.types';

export type CommonDateProps = {
  onDatePick: (date: Nullable<Date>) => void;
  selectedDate: Nullable<Date>;
};
