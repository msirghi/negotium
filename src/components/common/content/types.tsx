export type TaskWrapperTitleOptions = {
  onSave: (newTitle: string) => void;
  title: string;
  inputFontSize?: number;
};
