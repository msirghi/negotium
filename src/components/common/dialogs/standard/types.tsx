export type DialogOptions = {
  cancelButtonLabel: string;
  submitButtonLabel: string;
  dialogTitle: string;
  infoMessage?: string;
};
