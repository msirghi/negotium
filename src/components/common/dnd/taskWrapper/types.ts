import {Task} from "../../../../common/types/tasks.types";

export type DndTaskWrapperProps = {
  handleDragEnd: (result: any) => void;
};

export type TaskOrderUpdateDto = {
  updatedOrderNumbers: Array<{
    id: Task['id'];
    orderNumber: number;
  }>;
};
