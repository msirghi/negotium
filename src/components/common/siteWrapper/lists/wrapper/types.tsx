export type SiteWrapperListOptions = {
  addOptions?: {
    onClick: () => void;
  };
};
