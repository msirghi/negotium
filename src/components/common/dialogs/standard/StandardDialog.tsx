import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
  Divider,
  Tooltip,
} from '@mui/material';
import { FC } from 'react';
import { DialogOptions } from './types';
import { makeStyles } from '@mui/styles';
import colors from '../../../../common/styles/colors';
import InfoOutlinedIcon from '@mui/icons-material/InfoOutlined';
import { Box } from '@mui/system';
import { useIsMobile } from '../../../../common/hooks/common/useIsMobile';

type Props = {
  open: boolean;
  onSubmit: () => void;
  setOpen: (val: boolean) => void;
  options: DialogOptions;
  hideBackdrop?: boolean;
};

const useStyles = makeStyles({
  buttons: {
    marginTop: 10,
  },
  titleContainer: {
    position: 'relative',
    backgroundColor: colors.greys['100'],
  },
  divider: {
    position: 'absolute',
    width: '100%',
    left: 0,
    marginTop: 15,
  },
  info: {
    position: 'absolute',
    cursor: 'pointer',
    right: '5%',
    top: '50%',
    transform: 'translate(-50%, -50%)',
  },
  content: {
    padding: '35px 0 20px 0',
  },
});

export const StandardDialog: FC<Props> = ({
  open,
  setOpen,
  options,
  onSubmit,
  children,
  hideBackdrop,
}) => {
  const { dialogTitle, cancelButtonLabel, submitButtonLabel, infoMessage } =
    options;
  const classes = useStyles();
  const isMobile = useIsMobile();

  const handleClose = () => {
    setOpen(false);
  };

  return (
    <Dialog open={open} onClose={handleClose} hideBackdrop={hideBackdrop}>
      <DialogTitle className={classes.titleContainer}>
        <div>{dialogTitle}</div>
        <Divider className={classes.divider} />
        {options.infoMessage && (
          <Tooltip title={options.infoMessage}>
            <InfoOutlinedIcon data-testid={'info-icon'} className={classes.info} fontSize={'small'} />
          </Tooltip>
        )}
      </DialogTitle>
      <DialogContent>
        <Box className={classes.content}>{children}</Box>
      </DialogContent>

      <DialogActions className={classes.buttons}>
        <Button
          fullWidth={isMobile}
          onClick={handleClose}
          data-testid={'cancel-button'}
          size={'small'}
        >
          {cancelButtonLabel}
        </Button>
        <Button
          size={'small'}
          variant={'contained'}
          data-testid={'save-button'}
          fullWidth={isMobile}
          onClick={onSubmit}
        >
          {submitButtonLabel}
        </Button>
      </DialogActions>
    </Dialog>
  );
};
