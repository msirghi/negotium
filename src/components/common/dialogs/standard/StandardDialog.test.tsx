import { DialogOptions } from './types';
import renderer from 'react-test-renderer';
import { StandardDialog } from './StandardDialog';
import { mount } from 'enzyme';
import { Dialog } from '@mui/material';
import { act, fireEvent, render } from '@testing-library/react';

describe('StandardDialog', () => {
  const defaultProps = {
    open: false,
    setOpen: jest.fn(),
    options: {
      submitButtonLabel: 'Submit',
      cancelButtonLabel: 'Cancel',
      dialogTitle: 'Title',
    } as DialogOptions,
    onSubmit: jest.fn(),
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', () => {
    const tree = renderer.create(<StandardDialog {...defaultProps} />);
    expect(tree).toMatchSnapshot();
  });

  it('should call prop method on dialog close', () => {
    const wrapper = mount(<StandardDialog {...defaultProps} />);
    const dialog = wrapper.find(Dialog) as any;
    act(() => {
      dialog.props().onClose({}, 'escapeKeyDown');
    });
    expect(defaultProps.setOpen).toBeCalled();
  });

  it('should render info icon if info message is provided', () => {
    const updOptions = { ...defaultProps.options };
    updOptions.infoMessage = 'message';

    const { getByTestId } = render(
      <StandardDialog {...defaultProps} options={updOptions} open />
    );
    expect(getByTestId('info-icon')).toBeInTheDocument();
  });

  it('should call onSubmit prop method on save button click', () => {
    const { getByTestId } = render(<StandardDialog {...defaultProps} open />);
    const button = getByTestId('save-button');
    act(() => {
      fireEvent.click(button);
    });
    expect(defaultProps.onSubmit).toBeCalled();
  });
});
