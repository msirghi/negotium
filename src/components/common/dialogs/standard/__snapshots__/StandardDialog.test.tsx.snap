// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StandardDialog should match the snapshot 1`] = `null`;
